marimo/_messaging/types.py

```diff
@@ -1,6 +1,39 @@
 # Copyright 2024 Marimo. All rights reserved.
-from typing import Any, Tuple
+import abc
+from typing import Any, Dict, Optional, Tuple
+
+from marimo._ast.cell import CellId_t
 
 # The message from the kernel is a tuple of message type
 # and a json representation of the message
 KernelMessage = Tuple[str, Any]
+
+
+class Stream(abc.ABC):
+    """
+    A stream is a class that can write messages from the kernel to
+    some output.
+    The `write` method is called by the kernel.
+    """
+
+    cell_id: Optional[CellId_t] = None
+
+    @abc.abstractmethod
+    def write(self, op: str, data: Dict[Any, Any]) -> None:
+        pass
+
+
+class Stdout:
```

Comment:
Since we're using this as the expected type in various places, let's add the other public methods that Stdout should have. Same for Stderr and Stdin. In particular the public methods that exist on ThreadsafeStd*